src/charm.py

```diff
@@ -176,6 +186,29 @@ def _on_cluster_relation_changed(self, event: EventBase) -> None:
         if self.state.cluster.switching_encryption and len(self.state.servers) == 1:
             event.defer()
 
+    def _on_secret_changed(self, event: SecretChangedEvent):
+        """Reconfigure services on a secret changed event."""
+        if not event.secret.label:
+            return
```

Comment:
**question:** Under what conditions does this happen?

Reply:
If your charm at any point just creates a secret w/o label. Or if a related charm may share (`secret.grant()`) a secret with your charm.

Thus your charm is an observer of that secret that doesn't have a label -- and gets `secret-changed` event notification once that secret changes.

We don't care about any of those potential secrets.

NOTE: `label` is an OPTIONAL "sticker" we stick on secrets, so we have a way to identify them. As without, secret identification becomes... complicated.

Reply:
I'm assuming that the labelling of secrets is something we handle in our lib then?